File: CS/GrowthWare.Framework/Model/Profiles/Interfaces/IMSecurityInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GrowthWare.Framework.Model.Profiles.Interfaces
{
    /// <summary>
    /// Ensures the basic properties are avalible to all Profile model objects.
    /// </summary>
    /// <remarks>
    /// If it is decided to use entities in the future then
    /// this interface should be used for the save, delete, and getitem methods.
    ///  </remarks>
    [CLSCompliant(true)]
    public interface IMSecurityInfo
    {
        string[] AddRoles
        {
            get;
        }
        string[] DeleteRoles
        {
            get;
        }
        string[] EditRoles
        {
            get;
        }
        string[] ViewRoles
        {
            get;
        }
    }
}

File: VB/GrowthWare.Framework/Model/Profiles/Interfaces/IMSecurityInfo.vb
Namespace Model.Profiles.Interfaces
    <CLSCompliant(True)> _
    Public Interface IMSecurityInfo
        ReadOnly Property AddRoles() As String()
        ReadOnly Property DeleteRoles() As String()
        ReadOnly Property EditRoles() As String()
        ReadOnly Property ViewRoles() As String()
    End Interface
End Namespace
